pythoncz_tests/models/github_test.py

```diff
@@ -0,0 +1,432 @@
+
+    assert formatted_issue['votes'] == 42 + 3
+
+
+def test_calculate_votes(github):
+    """
+    The '_calculate_votes' helper should correctly deal with negative votes
+    """
+    votes = github._calculate_votes(fixtures.issue(reactions_counts={
+        'THUMBS_UP': 1,
+        'THUMBS_DOWN': 1,
+        'LAUGH': 1,
+        'HOORAY': 1,
+        'CONFUSED': 1,
+        'HEART': 1,
+    }))
+
+    assert votes == 2
+
+
+def test_get_nodes_without_total_count(github):
+    subnodes = [1, 2, 3, 4]
+    result = github._get_nodes({'something': {'nodes': subnodes}}, 'something')
+
+    assert result == subnodes
+
+
+def test_get_nodes_with_total_count(github):
+    subnodes = [1, 2, 3, 4]
+    result = github._get_nodes({'something': {
+        'totalCount': len(subnodes),
+        'nodes': subnodes,
+    }}, 'something')
+
+    assert result == subnodes
+
+
+def test_get_nodes_with_different_total_count(github):
+    """
+    If the total count of sub-nodes is different then the number of sub-nodes
+    available in the API response, at least warn about the fact that there's
+    some data missing and limits should be raised or results paginated
+    """
+    subnodes = [1, 2, 3, 4]
+    node = {'something': {
+        'totalCount': len(subnodes) + 42,
+        'nodes': subnodes,
+    }}
+    with warnings.catch_warnings(record=True) as recorded_warnings:
+        result = github._get_nodes(node, 'something')
+
+    assert result == subnodes
+    assert len(recorded_warnings) == 1
+    assert issubclass(recorded_warnings[0].category, UserWarning)
+    assert 'nodes in total, but only' in str(recorded_warnings[0].message)
+
+
+def test_sort_issues_coach(github):
+    """
+    Issues with coaching offer should always go first no matter what
+    """
+    repository = fixtures.repository()
+    issues = [github._format_issue('org', repository, issue) for issue in [
+        fixtures.issue(labels=[]),
+        fixtures.issue(labels=[{'name': 'foo'}]),
+        fixtures.issue(labels=[{'name': 'bar'}, {'name': 'coach'}]),
+        fixtures.issue(labels=[]),
+    ]]
+    sorted_issues = github._sort_issues(issues)
+
+    assert sorted_issues[0]['coach'] is True
+
+
+def test_sort_issues_votes(github):
+    """
+    Issues with most votes should go first if there's no 'coach' issue
+    """
+    repository = fixtures.repository()
+    issues = [github._format_issue('org', repository, issue) for issue in [
+        fixtures.issue(labels=[], reactions_counts={'LAUGH': 1}),
+        fixtures.issue(labels=[], reactions_counts={'THUMBS_DOWN': 1}),
+        fixtures.issue(labels=[], reactions_counts={'HEART': 5, 'CONFUSED': 1}),  # NOQA
+        fixtures.issue(labels=[], reactions_counts={'LAUGH': 1}),
+    ]]
+    sorted_issues = github._sort_issues(issues)
+
+    assert sorted_issues[0]['votes'] == 4
+
+
+def test_sort_issues_activity(github):
+    """
+    Issues with most user activity should go first if there's no 'coach'
```

Comment:
It's not clear from the docstring that this is sorted by the sum of comment count and participants count. Also make sure that comments only would sort differently (because here `3 > 1 > 0`), so the test would pass if the only criterion would be comments count.